src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -616,6 +617,34 @@ private void verifyNoHeaderJunk(TaskListener listener) throws IOException, Inter
         }
     }
 
+    private void interrogateRemoteSystem(TaskListener listener) throws IOException, InterruptedException {
```

Comment:
please, Could you add a javaDoc description?

Reply:
I'd prefer to add two new settings to the Agent configuration, auto detections based on commands outputs use to change with the time and different OS versions could return different results, a setting is always a setting.